pyquil/wavefunction.py

```diff
@@ -147,21 +148,34 @@ def pretty_print_probabilities(self, decimal_digits=2):
                 outcome_dict[outcome] = prob
         return outcome_dict
 
-    def pretty_print(self, decimal_digits=2):
+    def pretty_print(self, decimal_digits=2, ancillae: Optional[Sequence[int]] = None):
         """
         Returns a string repr of the wavefunction, ignoring all outcomes with approximately zero
         amplitude (up to a certain number of decimal digits) and rounding the amplitudes to
         decimal_digits.
 
         :param int decimal_digits: The number of digits to truncate to.
+        :param ancillae: The list of indices for ancillary qubits to omit.
         :return: A dict with outcomes as keys and complex amplitudes as values.
         :rtype: str
         """
+
+        # Initialize bitmask as 1's complement of ancilla spec
+        if ancillae is None:
+            ancillae = []
+        mask = 0
+        for ancilla in ancillae:
+            mask |= 1 << ancilla
+
         outcome_dict = {}
         qubit_num = len(self)
         pp_string = ""
         for index, amplitude in enumerate(self.amplitudes):
+            if index & mask != 0 and not np.isclose(amplitude, 0):
+                raise ValueError("Specified ancilla was found to have non-zero amplitude.")
             outcome = get_bitstring_from_index(index, qubit_num)
+            for ancilla in ancillae:
+                outcome = outcome[0 : qubit_num - ancilla - 1] + outcome[qubit_num - ancilla :]
```

Comment:
I'm not in favour of the spacing around the `:`. Does the style checker/formatter cause this?

Reply:
Yeah, that's the only way it passes `make check-format`

Reply:
errrrrghhhhh